### PR TITLE
Add custom Babel config and docgen-loader

### PR DESCRIPTION
**Pull request by mrmckeb**
This work is a continuation of #25.

See #36 for details.

### CODE REVIEW

**Comment by mrmckeb** [comment]
@shilman, I'm having trouble getting `react-docgen-typescript-loader` to work here. It seems there are a number of related issues at the moment:
https://github.com/strothj/react-docgen-typescript-loader/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+babel-loader

I'll try again when I'm in London...

**Comment by mrmckeb** [comment]
@shilman, as mentioned, I'm still having an issue with `react-docgen-typescript-loader`. It only works for some props, and I'm not sure why (yet). I think it's related to https://github.com/storybookjs/storybook/issues/8114.

File: packages/preset-create-react-app/helpers/processCraConfig.js [hunk 1]
```diff
@@ -22,12 +34,54 @@ const processCraConfig = (craWebpackConfig, storybookConfigDir) =>
...
+                    loader,
+                    options: {
+                      ...options,
+                      extends: _extends,
+                      plugins: [...plugins, ...options.plugins],
+                      presets: [...presets, ...options.presets],
+                    },
+                  },
+                  {
+                    loader: require.resolve('react-docgen-typescript-loader'),
```
Comment (shilman) [inline packages/preset-create-react-app/helpers/processCraConfig.js]: move this, scope to TSX OR even make it opt-in?

Comment (shilman) [reply packages/preset-create-react-app/helpers/processCraConfig.js]: docgenTypesriptLoaderOptions: null | { ... }

**Comment by shilman** [comment]
Remove DS_STORE and add to .gitignore

**Pull request merged**